### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by ranjanan**

### CODE REVIEW

**Comment by ranjanan** [comment]
@tkelman does this look alright?

File: test/runtests.jl [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+cd = CuArray(c)
+CUDA.launch(vdiv, 10, 1, (ad,bd,cd))
+c = to_host(cd)
+@test_approx_eq c a./b 
+
+#Negative test cases
+a = rand(Float32, 10)
+ad = CuArray(Float32, 5)
+try 
+	copy!(ad, a)
```
Comment (tkelman) [inline test/runtests.jl]: stick with 4 space indent, as that's used elsewhere in the package and the base julia style conventions

Comment (tkelman) [inline test/runtests.jl]: also if you expect this to always throw, add a catch all `error("should not get here")` after the line that is expected to fail, but before the catch

Comment (tkelman) [inline test/runtests.jl]: or just use test_throws

**Comment by ranjanan** [comment]
Alright, thanks. @ViralBShah , can we merge this?

**Pull request merged**